app/services/segment/response_middleware.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Segment
+  class ResponseMiddleware < Faraday::Response::Middleware
```

Comment:
When version `1.0` of `Faraday` is released, we won't need this anymore because it will be implemented there thanks to https://github.com/lostisland/faraday/pull/841
But we are currently using `0.9.2`, and as that PR says, `ClientError` in this version means almost any kind of error, including errors from the server side (`5XX`), and we need to differentiate to retry only the server ones 😄

Reply:
Do you think it is feasible to upgrade faraday soon?

Reply:
No idea 🤔😣 It doesn't depend much on us. Whenever they release it.